Automattic/jetpack
# JS Components: Add ProductOffer component

This PR adds the `<ProductDetail />` component to jetpack-components pkg.

* Based on RNA components: Text, H3, Button, etc
* Sizes, positions based on `--spacing-base` unit

Fixes https://github.com/Automattic/jetpack/issues/23843

#### Changes proposed in this Pull Request:

* JS Components: Add ProductDetailCard component 

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* With Storybook
* Confirm you can see the components
* Play with the story controls
* Check the box-model, based on the image shown in the issue description

![image](https://user-images.githubusercontent.com/77539/162189118-3c1cfc7b-2010-4f93-800c-717f86548d16.png)

<img width="647" alt="image" src="https://user-images.githubusercontent.com/77539/162189606-3a8b66b5-68e7-454e-aa4f-bf88edd90ed9.png">



https://user-images.githubusercontent.com/77539/162189956-70b622f1-73ff-476b-ba99-853356e5e0f9.mov
